MapboxNavigation/SpeechSynthesizerController.swift

```diff
@@ -0,0 +1,85 @@
+
+import Foundation
+import MapboxDirections
+import MapboxCoreNavigation
+
+public protocol SpeechSynthesizerController: class {
+    typealias SpeechSynthesizerCompletion = (Error?) -> Void
+    
+    ///
+    var delegate: SpeechSynthesizerDelegate? { get set }
+    
+    ///
+    var muted: Bool { get set }
+    ///
+    var volume: Float { get set }
+    ///
+    var isSpeaking: Bool { get }
+    ///
+    var locale: Locale { get set }
+    
+    ///
+    func changedIncomingSpokenInstructions(_ instructions: [SpokenInstruction])
+    ///
+    func speak(_ instruction: SpokenInstruction, completion: SpeechSynthesizerCompletion?)
```

Comment:
Following what was mentioned in original ticket, I don't feel like such protocol should have API to control `ducking` because:
1. It is responsibility of a `SpeechSynthesizer` to know and control when it needs to `duck`, if it needs at all
2. `Ducking` feature is performed via system methods and does not depend on particular synthesizer implementation.